app.py

```diff
@@ -140,24 +84,95 @@ def message(event, client):
-            # initializes the file for the messages and puts cursor at the last line
-            file = open("messages.txt", "a+")
-            file.write("Title of Documentation: {}\n".format(text))
-            file.close()
-    except TypeError:
-        return
+    # sends the onboarding message if "start" is typed.
+    try:
+        if text.lower() == "start":
+            return start_onboarding(user_id, channel_id, client)
+    except AttributeError:
+        None
+
+    # Scrapes message into the terminal; used for debugging purposes.
+    logger.info(f"New Message: {text}")
+
+    # The code below asks supplemental questions (except the title) and records user replies.
+    # ========================= Code Start ========================= #
+
+    # if the message is in a thread, proceed; if not, ignore.
+    if message_thread_ts != None:
+        # has_round_pushpin checks whether the parent message of the thread has a round pushpin reaction and returns a boolean.
+        # the variable uses the reactions.get method, which retrieves more than just the names of the reactions on the message. 
+        # for more information: https://api.slack.com/methods/reactions.get 
+        has_round_pushpin = "round_pushpin" in str(client.reactions_get(channel=channel_id, timestamp=message_thread_ts).get("message", {}).get("reactions", {}))
+
+        # if the parent message has a round pushpin reaction, proceed; if not, ignore.
+        if has_round_pushpin:
+            # if the parent message has not been documented, proceed; if not, ignore.
+            if not message_in_documentation(message_thread_ts):
+                # creates temporary variables to store whether the bot has asked a certain question.
+                question_1_asked, question_2_asked, question_3_asked = False, False, False
+
+                # Calls conversations.replies to get the thread history AND the message intended to be scraped.
+                # for more information on the conversation.replies method: https://api.slack.com/methods/conversations.replies
+                thread_message_history = client.conversations_replies(
+                    channel=channel_id,
+                    inclusive=True,
+                    ts=message_thread_ts,
+                    latest=message_ts
+                )["messages"]
+
+                # compares the thread message history to the individual questions. 
+                # if a the exact string of a question is in message["text"], mark corresponding boolean as true.
+                for message in thread_message_history:
+                    if message["text"] == onboarding_tutorial.question_one:
+                        question_1_asked = True
+                    elif message["text"] == onboarding_tutorial.question_two:
+                        question_2_asked = True
+                    elif message["text"] == onboarding_tutorial.question_three:
+                        question_3_asked = True

```

Comment:
You might consider using a single integer variable to track which question you're on, but up to you.

Reply:
Thanks - I didn't think of that. Using an integer variable definitely makes the code a lot more readable & is also easier to scale up if more questions were to be added.

A single integer variable has replaced the 3 Boolean variables to track what questions the bot has asked.